src/css/angle.zig

```diff
@@ -0,0 +1,92 @@
+const std = @import("std");
+const Parser = @import("parser.zig").Parser;
+const expectParse = @import("parser.zig").expectParse;
+
+/// A shared type for `angle`, `angle-percentage`
+pub const Angle = union(enum) {
+    // https://www.w3.org/TR/css-values-4/#absolute-lengths
+    deg: f32,
+    grad: f32,
+    rad: f32,
+    turn: f32,
+
+    // should be in a separate type but this is easier
+    percent: f32,
+
+    pub fn format(self: Angle, comptime _: []const u8, _: std.fmt.FormatOptions, writer: anytype) !void {
+        return switch (self) {
+            .percent => |v| writer.print("{d}%", .{v}),
+            inline else => |v| writer.print("{d}{s}", .{ v, @tagName(self) }),
+        };
+    }
+
+    pub fn parse(parser: *Parser) !Angle {
+        switch (try parser.tokenizer.next()) {
+            .number => |n| return .{ .deg = n },
+            .dimension => |d| inline for (std.meta.fields(Angle)) |f| {
+                if (comptime f.type == f32) {
+                    if (std.mem.eql(u8, d.unit, "percent")) {
```

Comment:
I think this condition is unnecessary, `percent` is also type of `f32` and it's a field in `Angle` so it would be checked and initialized with `@unionInit` expression